tests/regression/09-regions/38-pool.c

```diff
@@ -0,0 +1,263 @@
+// PARAM: --set ana.activated[+] region --set ana.activated[+] symb_locks
```

Comment:
Was it regions or symb_locks that was really required for this? Because currently this example is in the regions directory.

Reply:
both to get to zero warnings

Reply:
With symb_locks activated and regions deactivated, there's a race on the `threadLimit` field:
```
[Warning][Race] Memory location (struct POOL_ctx_s).threadLimit (race with conf. 100):
  read with [symblock:{p-lock:*.queueMutex}, mhp:{tid=[main], {POOL_thread@tests/regression/09-regions/38-pool.c:221:13-225:9}}] (conf. 100)  (exp: & ctx->threadLimit) (tests/regression/09-regions/38-pool.c:149:17-150:58)
  read with [symblock:{p-lock:*.queueMutex}, mhp:{tid=[main, POOL_thread@tests/regression/09-regions/38-pool.c:221:13-225:9]}, thread:[main, POOL_thread@tests/regression/09-regions/38-pool.c:221:13-225:9]] (conf. 100)  (exp: & ctx->threadLimit) (tests/regression/09-regions/38-pool.c:149:17-150:58)
  write with [mhp:{tid=[main]; created={[main, POOL_thread@tests/regression/09-regions/38-pool.c:221:13-225:9]}}, thread:[main]] (conf. 100)  (exp: & ctx->threadLimit) (tests/regression/09-regions/38-pool.c:227:9-227:38)
```
And I think that's actually a valid race: the threads read it with the mutex, but the write is completely unlocked, no?

Therefore I think this is another example where our region analysis unsoundly assigns `no region` (bullet) to something which actually can be shared (by passing the pointer to the thread via its create argument).